frame/clearing-house/src/lib.rs

```diff
@@ -917,6 +1003,38 @@ pub mod pallet {
 			Self::deposit_event(Event::FundingUpdated { market: market_id.clone(), time: now });
 			Ok(())
 		}
+
+		#[transactional]
+		fn liquidate(
+			liquidator_id: &Self::AccountId,
+			user_id: &Self::AccountId,
+		) -> Result<(), DispatchError> {
+			let summary = Self::summarize_account_state(user_id)?;
+
+			let fees = Self::fully_liquidate_account(user_id, summary)?;
+
+			// Charge fees
+			let liquidator_fee =
+				Self::full_liquidation_penalty_liquidator_share().saturating_mul_int(fees);
```

Comment:
In `saturating_mul_int` [doc](https://paritytech.github.io/substrate/master/sp_runtime/trait.FixedPointNumber.html#method.saturating_mul_int) it says that the result could be either `N::min` or `N::max`.

I didn't fully understand why it would return `N::min` neither why it would be desirable here, could you please help me here? :sweat_smile:

Reply:
`saturating_mul_int` can operate on either signed or unsigned values, so the docs are general in that sense. In this case, `full_liquidation_penalty_liquidator_share()` is a fixed point number in `[0, 1]` (this will actually be enforced when setters for the Clearing House parameters are implemented) and `fees` is a balance, so the expected result should be in `[0, fees]`

Reply:
Oh I see. 
Thanks for the explanation :+1: